sigopt/xgboost/run.py

```diff
@@ -312,17 +313,19 @@ def log_validation_metrics(self):
       if self.early_stopping_rounds:
         self.run.log_metric('num_boost_round_before_stopping', n_eval_rounds)
 
-    if self.run_options_parsed['autolog_metrics']:
-      if self.validation_sets:
-        for validation_set in self.validation_sets:
-          if self.is_regression:
-            self.run.log_metrics(
-              compute_regression_metrics(self.model, (validation_set))
-            )
-          else:
-            self.run.log_metrics(
-              compute_classification_metrics(self.model, (validation_set))
-            )
+    if not self.run_options_parsed['autolog_metrics']:
+      return
+
+    if self.validation_sets:
```

Comment:
could save a few lines and just do
`if self.run_options_parsed['autolog_metrics'] and self.validation_sets:`

idk why the previous code had two nested if statements like that, they can be combined into one